.github/workflows/wallet_build.yml

```diff
@@ -1,47 +1,72 @@
-name: compile_wallet_binaries
-
-on:
-  workflow_dispatch:
-
-defaults:
-  run:
-    shell: bash
-
-jobs:
-  build_and_publish:
-    name: Build rusk-wallet binaries for ${{ matrix.os }} with ${{ matrix.compiler }}.
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest]
-        compiler: [cargo]
-        include:
-          - os: ubuntu-latest
-            compiler: cargo
-            target: linux
-
-          - os: macos-latest
-            compiler: cargo
-            target: macos
-
-    steps:
-      - name: Checkout Repository
-        uses: actions/checkout@v2
-
-      - name: Install dependencies
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-      - run: rustup component add rustfmt
-
-      - name: Build Rusk binary
-        shell: bash
-        working-directory: ./rusk
-        run: ${{matrix.compiler}} b --release --verbose -p rusk-wallet
-
-      - name: "Upload Rusk Artifact"
-        uses: actions/upload-artifact@v2
-        with:
-          name: rusk-wallet
-          path: ./target/release/rusk-wallet
-          retention-days: 5
+name: Compile CLI wallet binaries
+
+on:
+  workflow_dispatch:
+    inputs:
+      dusk_blockchain_ref:
+        description: "GIT branch to checkout"
+        required: true
+        default: "master"
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build_and_publish:
+    name: Build rusk-wallet binaries for ${{ matrix.os }} with ${{ matrix.compiler }}.
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        compiler: [cargo]
+        include:
+          - os: ubuntu-latest
+            compiler: cargo
+            target: linux-x64
+
+          - os: macos-latest
+            compiler: cargo
+            target: macos-intel
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.dusk_blockchain_ref }}
+
+      - name: Install dependencies
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+
+      - name: Add rustfmt component
+        run: rustup component add rustfmt
+
+      - name: Build Rusk binary
+        shell: bash
+        working-directory: ./rusk
+        run: ${{matrix.compiler}} b --release --verbose -p rusk-wallet
+
+      - name: Get semver from wallet binary
+        run: |
+          export SEMVER=$(./target/release/rusk-wallet --version | perl -lpe 's/\w+\s\w+\s\w+\s(.*?)/$1/')
```

Comment:
As discussed this is a bit fragile since we could change the naming of the wallet (as most likely we'll do, "Dusk Wallet CLI" it's a bit wrong, especially the "CLI" part). In order for you to grab the version, the most solid approach is taking from the [cargo.toml](https://github.com/dusk-network/rusk/blob/master/rusk-wallet/Cargo.toml#L3).

As alternative, we could provide in our binaries an option to printout just the version in semver format without the name and build as we usually do.